tests/feature-store/test_feature_store.py

```diff
@@ -44,10 +44,76 @@ def test_basic_featureset():
     assert (
         stocks_set.spec.features["name"].description == "some name"
     ), "description was not set"
-    assert len(df) == len(stocks), "datafreame size doesnt match"
+    assert len(df) == len(stocks), "dataframe size doesnt match"
     assert stocks_set.status.stats["exchange"], "stats not created"
 
 
+def verify_ingest(
+    base_data, keys, infer=False, targets=None, infer_options=fs.InferOptions.default()
+):
+    if isinstance(keys, str):
+        keys = [keys]
+    feature_set = fs.FeatureSet("my-feature-set")
+    if infer:
+        data = base_data.copy()
+        fs.infer_metadata(feature_set, data, entity_columns=keys)
+    else:
+        data = base_data.set_index(keys=keys)
+    if targets:
+        feature_set.set_targets(targets=targets, with_defaults=False)
+    df = fs.ingest(feature_set, data, infer_options=infer_options)
+
+    assert len(df) == len(data)
+    if infer:
+        data.set_index(keys=keys, inplace=True)
+    for idx in range(len(df)):
+        assert all(df.values[idx] == data.values[idx])
+
+
+@pytest.mark.skipif(not has_db(), reason="no db access")
+def test_featureset_column_types():
+    init_store()
+
+    data = pd.DataFrame(
+        {
+            "key": ["key1", "key2"],
+            "str": ["my_string1", "my_string2"],
+            "int": [123456, 234567],
+            "float": [123.456, 234.567],
+            "bool": [True, False],
+            "timestamp": [
+                pd.Timestamp("1980-02-04 17:21:50.781"),
+                pd.Timestamp("2020-03-04 12:12:45.120"),
+            ],
+            "category": pd.Categorical(
+                ["a", "c"], categories=["d", "c", "b", "a"], ordered=True
+            ),
+        }
+    )
+    verify_ingest(data, "key")
+    verify_ingest(data, "key", infer=True)
+    verify_ingest(data, "str")
+    verify_ingest(data, "str", infer=True)
+    verify_ingest(data, "int")
+    verify_ingest(data, "int", infer=True)
+    verify_ingest(data, "float")
+    verify_ingest(data, "float", infer=True)
+    verify_ingest(data, "bool")
+    verify_ingest(data, "bool", infer=True)
+    verify_ingest(data, "timestamp")
+    verify_ingest(data, "timestamp", infer=True)
+    verify_ingest(data, "category")
+    verify_ingest(data, "category", infer=True)
```

Comment:
I see you're already supporting keys to be a list in `verify_ingest`, why not using it and give one key every time ?
Even if there is some reason for that, this code can be written in much more elegant way, e.g.:
```
for key in data.keys():
    verify_ingest(data, key, infer=False)
    verify_ingest(data, key, infer=True)
```

Reply:
It's supported as feature store should support multi column keys (in progress).